### ISSUE TITLE
Replace Dialog: "Some of the files cannot be modified." gets longer than screen height ...

### ISSUE THREAD

**Issue by Dromantor** [dnGrep/dnGrep#1123]
![grafik](https://github.com/dnGrep/dnGrep/assets/1822013/943f976a-2635-4db8-9429-d29a8c7150e7)

Btw: What's so special to windows .lnk files that they cannot be modified? I can change any binary file I want, but not .lnk? (I changed the drive letter of my data drive, many files have to be changed now, registry is already done.)

**Comment by doug24** [comment]
What other binary file format does dnGrep let you change?  They should all be blocked - it is not supported. From the documentation:

> dnGrep does not replace in binary file formats such as Word, Excel, PowerPoint, PDF, ebooks, files in archives, images, audio files, or executable files.

**Comment by Dromantor** [comment]
Are binary files detected by a list of file extensions or the content? If it's the second one some files slip through that detection. In the search i've open right now a png (without file extension) and a .xpi (some language pack file from thunderbird) isn't read only.

**Comment by Dromantor** [comment]
Yesterday so many files weren't r/o I didn't even notice this was a thing.

**Comment by doug24** [comment]
Binary files are detected - maintaining a list would be an impossible task.  Yes, no doubt the binary detection can be wrong either positive or negative - there just isn't any definitive way test that I know of.

Nothing should have changed with respect to files being read only in recent releases.

**Comment by Dromantor** [comment]
_Just writing some answer for the sake of answering, already wanted to delete everything again, so feel free to ignore._

I mean, at what point is it really a binary file? I sometimes have to do with config files that are 95% normal text and some binary as separators (or sth. like that; typical proprietary software stuff ...). One could r/o everything that contains unprintable or invalid chars ... but `A4+~KaNAS:\7,dsjkJKa$sd%8""34o!SADkl` doesn't contain any and could still be binary data ...
I've the solution: AI! ;) What, you don't have a RTX4090 to run the AI model quickly? Too bad, just come back in 8 hours ;-)

tl;dr: Doesn't matter, there will always be detection issues. I just want to see my popup dialog buttons ;-)

**Comment by doug24** [comment]
I think it's pretty obvious that a .lnk file is binary.  It is not an error in detection.  The files contain text in both single byte and multibyte encoding - how do you deal with that?  Not allowing replace in binary files is not an arbitrary restriction.  Good luck if you find another app that will allow you to do this.

**Comment by Dromantor** [comment]
Sorry, just wanted to be funny. Didn't work out.

To be clear: I know .lnk is binary, I know that you don't support binary files, I know that it is not an error in detection in this case, I don't want you application supports this just because I say so. Sorry if it came across this way. I already solved my case with a python script a few minutes after i did the original post.

All my bug report is about is the dialog box, the rest is unnecessary context and (not-so-)fun (it seems).
